### DISCUSSION TITLE
change channel description and name

### DISCUSSION THREAD

**Discussion by AnelHodzic** [Rapptz/discord.py#9056] [Q&A] [answered]
Goal is to change channel description and name directly from python code. With the code mentioned bellow I'm able to change name but for some reason description wont update. What am I missing here.

```
import discord
from discord.ext.commands import Bot
import datetime
from dotenv import load_dotenv  
import os
load_dotenv() 

TOKEN = os.getenv('DISCORD_TOKEN')
 
bot = discord.Bot("!", intents=discord.Intents.all())

@bot.event
async def on_ready():
    print(f'{bot.user} has connected to Discord!')
    bot_channel = bot.get_channel(int(os.getenv('CHANNEL_ID')))
    await bot_channel.edit(name = 'ChannelName'
        , description = 'ChannelDescription')
bot.run(TOKEN)
```

For some reason changing the channel description doesn't work at all, even so according to documentation I should be able to change it.  [GUILD EDIT](https://discordpy.readthedocs.io/en/latest/api.html#discord.Guild.edit)

`discord.py 2.0.1`

**Comment by tailoric** [comment, marked as answer]
I feel like you went to the wrong part of the documentation, you are linking to guild edit docs when you are talking about editing a channel and trying that in code too. If you check the [TextChannel docs](https://discordpy.readthedocs.io/en/stable/api.html?highlight=textchannel#discord.TextChannel.edit) 
you will see that there is no `description` parameter what you want to use is `topic`

**Reply by AnelHodzic** [reply]
Oh boy is that so obvious. :( .. I have a bonus question if its my happy hour :)

```
channel_id = 0

class DiscordNameChannel:
    def __init__(self, channel_id):
        self.channel_id = channel_id
    
    @bot.event
    async def on_ready():
        print(f'{bot.user} has connected to Discord!')
        bot_channel = bot.get_channel(int(channel_id))
        now = datetime.now() # current date and time

        date_time = now.strftime("%m/%d/%Y, %H:%M:%S")

        await bot_channel.edit(name = '🎁Synced', topic=date_time )
                
    def execute(self):
        bot.run(TOKEN)
 channel_id = xx
 myDiscord = DiscordNameChannel(channel_id)
 myDiscord.execute()

myOtherCode should be here. But it never comes to it 
```
Goal is just to make a function which will make me a change of topic and set it to current date time. 

Basically i have task which completes everything in background and i just need to call a function which will connect bot change server topic and log out. 

Anyway thank you for your help. I appreciate it

**Reply by tailoric** [reply]
First of all 
you should not use on_ready() for this.
on_ready can get called multiple times and editing a channel is heavily ratelimited 
If your whole intention is to only run the script update channel and log out you should instead use 
[`asyncio.create_task()`](https://docs.python.org/3/library/asyncio-task.html?highlight=asyncio%20create_task#asyncio.create_task) within either a dedicated `async def main` or within a `setup_hook`, check the following documentation https://discordpy.readthedocs.io/en/latest/migrating.html?highlight=setup_hook#asyncio-event-loop-changes for that. And inside your coroutine use [`await bot.wait_until_ready()`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Client.wait_until_ready) before getting and editing the channel and then call [`bot.close()`](https://discordpy.readthedocs.io/en/latest/api.html?highlight=close#discord.Client.close) at the end.

The problem with your current code is, that you overwrite the `on_ready` event inside a class instead of the global scope.
If you want to overwrite it inside a dedicated class you should subclass `commands.Bot` or `Client` and just create a function called `on_ready` (without decorator) within there. But again this is not recommended, instead use `setup_hook` and create a task with `asyncio.create_task` in there.
